### PR TITLE
[LCM] add: locm docs.

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: docs/source/en/using-diffusers/lcm.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+> LCMs can be distilled from any pre-trained Stable Diffusion (SD) in only 4,000 training steps (~32 A100 GPU Hours) for generating high quality 768 x 768 resolution images in 2~4 steps or even one step , significantly accelerating text-to-image generation. We employ LCM to distill the Dreamshaper-V7 version of SD in just 4,000 training iterations.
+
+For a more technical overview of LCMs, refer to [the paper](https://huggingface.co/papers/2310.04378).
+
+This guide shows how to perform inference with LCMs for text-to-image and image-to-image generation tasks. It will also cover performing inference with LoRA checkpoints.
+
+## Text-to-image
+
+You'll use the [`StableDiffusionXLPipeline`] here changing the `unet`. The UNet was distilled from the SDXL UNet using the framework introduced in LCM. Another important component is the scheduler: [`LCMScheduler`]. Together with the distilled UNet and the scheduler, LCM enables a fast inference workflow overcoming the slow iterative nature of diffusion models.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/lcm.md:27]: This is interesting, but is it not possible to use a LCM checkpoint directly, instead of replacing the UNet?

File: docs/source/en/using-diffusers/lcm.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+image = pipe(
+    prompt=prompt, num_inference_steps=4, generator=generator, guidance_scale=1.0
+).images[0]
+```
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/lcm/lcm_intro.png)
+
+Notice that we use only 4 steps for generation which is way less than what's typically used for standard SDXL.
+
+So notes to keep in mind:
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/lcm.md:55]:
```suggestion
Some details to keep in mind:
```

File: docs/source/en/using-diffusers/lcm.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+* For performing classifier-free guidance, usually batch size is doubled. When using LCM, classifier-free guidance is performed using guidance embeddings, so, the batch size is not doubled. This leads to a faster inference time. Since guidance is performed using guidance embeddings, negative prompts do not have any effect on the denoising process.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/lcm.md:57]:
```suggestion
* To perform classifier-free guidance, batch size is usually doubled inside the pipeline. LCM, however, applies guidance using guidance embeddings, so the batch size does not have to be doubled in this case. This leads to a faster inference time, with the drawback that negative prompts don't have any effect on the denoising process.
```

In the previous example we are using a standard `DiffusionPipeline`, how is that applying the guidance embeddings mentioned here (instead of the double batch size)?

File: docs/source/en/using-diffusers/lcm.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+* The UNet was trained using the [3., 13.] guidance scale range. So, that is the ideal range for `guidance_scale`. However, a `guidance_scale` of 1.0 is also effective.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/lcm.md:58]:
```suggestion
* The UNet was trained using the [3., 13.] guidance scale range. So, that is the ideal range for `guidance_scale`. However, disabling `guidance_scale` using a value of 1.0 is also effective in most cases.
```

File: docs/source/en/using-diffusers/lcm.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    guidance_scale=1,
+).images[0]
+```
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/lcm/lora_lcm.png)
+
+For LoRA, some of the above-mentioned findings change: 
+
+* By default, _no guidance is performed_ which enables faster inference. It works for most of the prompts. This means `guidance_scale` is set to 1 (as in the example above).
+* When `guidance_scale` > 1, regular classifier-free guidance is performed leading slower inference as this doubles the batch size.
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/lcm.md:191]:
```suggestion
* When `guidance_scale` > 1, regular classifier-free guidance is performed leading to slightly slower inference as this doubles the batch size. 
```

File: docs/source/en/using-diffusers/lcm.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+).images[0]
+```
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/lcm/lora_lcm.png)
+
+For LoRA, some of the above-mentioned findings change: 
+
+* By default, _no guidance is performed_ which enables faster inference. It works for most of the prompts. This means `guidance_scale` is set to 1 (as in the example above).
+* When `guidance_scale` > 1, regular classifier-free guidance is performed leading slower inference as this doubles the batch size. 
+* Recommended range for `guidance_scale` is [1., 2.].
```
Comment (pcuenca) [inline docs/source/en/using-diffusers/lcm.md:192]:
```suggestion
* The recommended range for `guidance_scale` is [1., 2.].
```

File: docs/source/en/using-diffusers/lcm.md (hunk 1 above)
Comment (sayakpaul) [reply docs/source/en/using-diffusers/lcm.md:27]: Wanted to explain that bit a bit explicitly.

File: docs/source/en/using-diffusers/lcm.md (hunk 3 above)
Comment (sayakpaul) [reply docs/source/en/using-diffusers/lcm.md:57]: I am using the `add_lcm_tests` branch. It takes care of it :)

File: docs/source/en/using-diffusers/lcm.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+## Text-to-image
+
+You'll use the [`StableDiffusionXLPipeline`] here changing the `unet`. The UNet was distilled from the SDXL UNet using the framework introduced in LCM. Another important component is the scheduler: [`LCMScheduler`]. Together with the distilled UNet and the scheduler, LCM enables a fast inference workflow overcoming the slow iterative nature of diffusion models.
+
+```python
+from diffusers import DiffusionPipeline, UNet2DConditionModel, LCMScheduler
+import torch
+
+unet = UNet2DConditionModel.from_pretrained(
+    "lcm-sd/lcm-sdxl-full-param-1k-huber-simian",
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:34]:
```suggestion
    "latent-consistency/lcm-sdxl",
```

File: docs/source/en/using-diffusers/lcm.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+pipe = DiffusionPipeline.from_pretrained(
+    "stabilityai/stable-diffusion-xl-base-1.0", unet=unet, torch_dtype=torch.float16
+).to("cuda")
+pipe.scheduler = LCMScheduler.from_config(pipe.scheduler.config)
+
+prompt = "Self-portrait oil painting, a beautiful cyborg with golden hair, 8k"
+
+generator = torch.manual_seed(0)
+image = pipe(
+    prompt=prompt, num_inference_steps=4, generator=generator, guidance_scale=1.0
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:47]:
```suggestion
    prompt=prompt, num_inference_steps=4, generator=generator, guidance_scale=8.0
```

For fully fine-tuned it's actually better to use higher guidance scale (sorry my messages were confusing before)

File: docs/source/en/using-diffusers/lcm.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+import torch
+from PIL import Image
+
+from diffusers.utils import load_image
+from diffusers import (AutoencoderKL, ControlNetModel, LCMScheduler,
+                       StableDiffusionXLControlNetPipeline,
+                       UNet2DConditionModel)
+
+unet = UNet2DConditionModel.from_pretrained(
+    "lcm-sd/lcm-sdxl-full-param-1k-huber-simian",
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:114]:
```suggestion
    "latent-consistency/lcm-sdxl",
```
```

File: docs/source/en/using-diffusers/lcm.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+    image=image,
+    num_inference_steps=4,
+    generator=generator,
+    guidance_scale=1.0,
+).images[0]
+```
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/lcm/lcm_i2i.png)
+
+## ControlNet
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:96]: Actually can we move this section to LoRA? ControlNet only works with the lora distilled yet (maybe show an example of v15 here): https://huggingface.co/latent-consistency/lcm-lora-sdv1-5

File: docs/source/en/using-diffusers/lcm.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+pipe = DiffusionPipeline.from_pretrained(model_id, variant="fp16", torch_dtype=torch.float16).to("cuda")
+
+pipe.load_lora_weights(lcm_lora_id, weight_name="lcm_sdxl_lora.safetensors")
+pipe.scheduler = LCMScheduler.from_config(pipe.scheduler.config)
+
+prompt = "close-up photography of old man standing in the rain at night, in a street lit by lamps, leica 35mm summilux"
+image = pipe(
+    prompt=prompt,
+    num_inference_steps=4,
+    guidance_scale=1,
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:182]:
```suggestion
    guidance_scale=0,
```

Actually `guidance_scale=0` also disables it and is cleaner.

File: docs/source/en/using-diffusers/lcm.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+).images[0]
+```
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/lcm/lora_lcm.png)
+
+For LoRA, some of the above-mentioned findings change: 
+
+* By default, _no guidance is performed_ which enables faster inference. It works for most of the prompts. This means `guidance_scale` is set to 1 (as in the example above).
+* When `guidance_scale` > 1, regular classifier-free guidance is performed leading to slightly slower inference as this doubles the batch size. 
+* The recommended range for `guidance_scale` is [1., 2.].
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:192]: Can we also show img2img and Controlnet - see: https://github.com/huggingface/diffusers/pull/5723/files#r1388130749

Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:192]:
```suggestion
* If `guidance_scale` is not disabled, the recommended range for `guidance_scale` is [1., 2.].
```

File: docs/source/en/using-diffusers/lcm.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+## LoRA
+
+It is possible to generalize the LCM framework to use with [LoRA](../training/lora.md). It effectively eliminates the need to conduct expensive fine-tuning runs as LoRA training concerns just a few number of parameters compared to full fine-tuning. During inference, the [`LCMScheduler`] comes to the advantage as it enables very few-steps inference without compromising the quality.
+
+```python
+from diffusers import DiffusionPipeline, LCMScheduler
+import torch
+
+model_id = "stabilityai/stable-diffusion-xl-base-1.0"
+lcm_lora_id = "lcm-sd/lcm-sdxl-lora-huber-final"
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/lcm.md:171]:
```suggestion
lcm_lora_id = "latent-consistency/lcm-lora-sdxl"
```

**Review by patrickvonplaten — COMMENTED** [review]
Very nice work! Added some updates regarding the naming :-) 

It would be great if we could add some more examples to the LoRA section, e.g.:
- Img2Img
- ControlNet (note that ControlNet doesn't work fully-fine-tuned lcm, but only for LoRA)
- Adding multiple LoRAs
- Img2Img
- Inpainting

File: docs/source/en/using-diffusers/lcm.md (hunk 10 above)
Comment (sayakpaul) [reply docs/source/en/using-diffusers/lcm.md:96]: But the result shows it works: 

image

File: docs/source/en/using-diffusers/lcm.md (hunk 11 above)
Comment (sayakpaul) [reply docs/source/en/using-diffusers/lcm.md:182]: @pcuenca tested it with 1.0.

Comment (patrickvonplaten) [reply docs/source/en/using-diffusers/lcm.md:182]: 0 and 1 is the same
